src/neo-vm/Utility.cs

```diff
@@ -25,6 +25,26 @@ static Utility()
             StrictUTF8.EncoderFallback = EncoderFallback.ExceptionFallback;
         }
 
+        public static BigInteger ModInverse(this BigInteger value, BigInteger modulus)
+        {
+            if (value <= 0) throw new ArgumentOutOfRangeException(nameof(value));
+            if (modulus < 2) throw new ArgumentOutOfRangeException(nameof(modulus));
+            BigInteger n = value, m = modulus, v = 0, d = 1;
+            while (n > 0)
+            {
+                BigInteger t = m / n, x = n;
+                n = m % x;
+                m = x;
+                x = d;
+                d = v - t * x;
+                v = x;
+            }
+            v %= modulus;
+            if (v < 0) v += modulus;
+            if (!(value * v % modulus).IsOne) throw new InvalidOperationException();
+            return v;
```

Comment:
```suggestion
            BigInteger r = value, old_r = modulus, old_s = 0, s = 1;
            while (r > 0)
            {
                BigInteger q = old_r / r;
                (old_r, r) = (r, old_r % r);
                (old_s, s) = (s, old_s - q * s);
            }
            old_s %= modulus;
            if (old_s < 0) old_s += modulus;
            if (!(value * old_s % modulus).IsOne) throw new InvalidOperationException();
            return old_s;
```

Reply:
For simplicity, just renamed `n` to `r`, `m` to `old_r`, `d` to `s` and `v` to `old_s`. I heard C#7 had these nice tuple assignments... hope this works.

Reply:
Updated.

Reply:
Very beautiful that this works on C#.